lb.tf

```diff
@@ -85,3 +85,30 @@ resource "aws_route53_record" "wildcard" {
 data "dns_a_record_set" "nlb" {
   host = module.nlb.lb_dns_name
 }
+
+
+locals {
+  # List of Node_Groups names from the module configuration which have "nlbs_attachment = true" or undefined (true by default)
+  node_groups = [for name, values in var.node_groups : name if lookup(values, "nlbs_attachment", false)]
+
+  # Map of "public" and/or "private" NLBs with their respective Target Groups ARNs
+  nlbs_target_groups = merge(
+    var.create_public_nlb ? { "public" = module.nlb.target_group_arns } : {},
+    var.create_private_nlb ? { "private" = module.nlb_private.target_group_arns } : {},
+  )
+
+  # Map of all Autoscaling Groups created by EKS managed or Self managed Node Groups, referenced by the Node Group name
+  eks_managed_autoscaling_groups_by_node_group  = { for item in setproduct(local.node_groups, module.cluster.eks_managed_node_groups_autoscaling_group_names) : item[0] => item[1] if startswith(item[1], "eks-${item[0]}") }
+  self_managed_autoscaling_groups_by_node_group = { for item in setproduct(local.node_groups, module.cluster.self_managed_node_groups_autoscaling_group_names) : item[0] => item[1] if startswith(item[1], item[0]) }
+  autoscaling_groups_by_node_group              = merge(local.eks_managed_autoscaling_groups_by_node_group, local.self_managed_autoscaling_groups_by_node_group)
+
+  # Map of all Autoscaling Groups to Target Groups attachments, with unique keys based only on data from module variables suitable for use in a for_each
+  autoscaling_attachments = { for item in setproduct(local.node_groups, keys(local.nlbs_target_groups), range(length(local.lb_target_groups))) : "${item[0]}_${item[1]}_${local.lb_target_groups[item[2]].backend_port}" => { "autoscaling_group" = local.autoscaling_groups_by_node_group[item[0]], "target_group_arn" = local.nlbs_target_groups[item[1]][item[2]] } }
+}
```

Comment:
```suggestion

locals {
  # List of node groups names from the module configuration which have "nlbs_attachment = true" or undefined (true by default)
  node_groups = [for name, values in var.node_groups : name if lookup(values, "nlbs_attachment", false)]

  # Map of "public" and/or "private" NLBs with their respective target group ARNs
  nlbs_target_groups = merge(
    var.create_public_nlb ? { "public" = module.nlb.target_group_arns } : {},
    var.create_private_nlb ? { "private" = module.nlb_private.target_group_arns } : {},
  )

  # Map of all autoscaling groups created by EKS managed or self-managed node groups, referenced by the node group name
  eks_managed_autoscaling_groups_by_node_group  = { for item in setproduct(local.node_groups, module.cluster.eks_managed_node_groups_autoscaling_group_names) : item[0] => item[1] if startswith(item[1], "eks-${item[0]}") }
  self_managed_autoscaling_groups_by_node_group = { for item in setproduct(local.node_groups, module.cluster.self_managed_node_groups_autoscaling_group_names) : item[0] => item[1] if startswith(item[1], item[0]) }
  autoscaling_groups_by_node_group              = merge(local.eks_managed_autoscaling_groups_by_node_group, local.self_managed_autoscaling_groups_by_node_group)

  # Map of all autoscaling groups to target groups attachments, with unique keys based only on data from module variables suitable for use in a for_each
  autoscaling_attachments = { for item in setproduct(local.node_groups, keys(local.nlbs_target_groups), range(length(local.lb_target_groups))) : "${item[0]}_${item[1]}_${local.lb_target_groups[item[2]].backend_port}" => { "autoscaling_group" = local.autoscaling_groups_by_node_group[item[0]], "target_group_arn" = local.nlbs_target_groups[item[1]][item[2]] } }
}
```

I know I'm just nitpicking on the comments, but I removed the capitalizations.

I checked the documentation from the upstream module and they are all over the place with the styling ([example](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples/self_managed_node_group)), so I think we are free to do what we want. In the end I removed them to be consistent to what we have in the README.adoc.

Reply:
BTW, thanks for the comments, future me will appreciate it when I come back to this code :wink:

Reply:
The idea of the capitalized names was to make the AWS resources we're talking about stand out. But I agree it looks a bit heavy.